### PR TITLE
Feature - add custom dependency tracking

### LINKED ISSUE #53: Provide capability to track a custom dependency
**Is your feature request related to a problem? Please describe.**
Provide capability to track a custom dependency for which we do not provide a convenience method.

This allows people to track dependencies that we don't support yet or are specific to their application.

**Describe the solution you'd like**
Provide capability to track a custom dependency with a new extension on `ILogger`.

```csharp
_logger.LogDependency("SendGrid", <uri>, startTime, durationMeasurement.Elapsed, telemetryContext);
```

Where uri is an object so that it can be anything maybe? open to suggestion.

It should allow you to define most of the info which the SDK allows you to define:
https://docs.microsoft.com/en-us/azure/azure-monitor/app/asp-net-dependencies#manually-tracking-dependencies

**Describe alternatives you've considered**
None

**Additional context**
We only support HTTP & SQL for now, but what if I want to track Service Bus, Table Storage, Twilio, SendGrid, etc.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Closes #53

### CODE REVIEW

File: docs/features/writing-different-telemetry-types.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ We provide support for the following dependencies:
 
 - [HTTP](#measuring-http-dependencies)
 - [SQL](#measuring-sql-dependencies)
+- [Custom](#measuring-custom-dependencies)
```
Comment (tomkerkhove) [inline docs/features/writing-different-telemetry-types.md]: Can you sort these alpahbetically please?

**Comment by tomkerkhove** [comment]
Do you have a sample of what this looks like in AI?

**Comment by stijnmoreels** [comment]
> Do you have a sample of what this looks like in AI?

not atm, maybe wait till #62  is merged and pull it here.

File: docs/features/writing-different-telemetry-types.md (hunk 1 above)
Comment (stijnmoreels) [reply docs/features/writing-different-telemetry-types.md]: Done but left the actual contents as it is (?).

**Comment by tomkerkhove** [comment]
Ok but you can also just run it locally

File: docs/features/writing-different-telemetry-types.md [hunk 2]
```diff
@@ -131,6 +131,48 @@ using (var measurement = DependencyMeasurement.Start("get-products"))
...
 
+### Measuring custom dependencies
+
+Here is how you can areport a custom depenency:
+
+
+```csharp
+var telemetryContext = new Dictionary<string, object>
+{
+    { "Local", "true" }
```
Comment (tomkerkhove) [inline docs/features/writing-different-telemetry-types.md:142]: Let's use a better example here for this demo:
```suggestion
    { "Subject", "Your order is being processed!" },
    { "OrderId", "ABC" }
```

File: docs/features/writing-different-telemetry-types.md [hunk 3]
```diff
@@ -131,6 +131,48 @@ using (var measurement = DependencyMeasurement.Start("get-products"))
...
+
+string dependencyName = "SendGrid";
+object dependencyData = "http://my.sendgrid.uri/"
+
+_logger.LogDependency("SendGrid", dependencyData, isSuccessful: true, startTime: startTime, duration: durationMeasurement.Elapsed, context: telemetryContext);
+// Output: "Dependency SendGrid http://my.sendgrid.uri/ in 00:00:01.2396312 at 03/23/2020 09:32:02 +00:00 (Successful: True - Context: [Local, True])"
+```
+
+Or alternatively, one can use our `DependencyMeasurement` model to manage timing for you:
+
```
Comment (tomkerkhove) [inline docs/features/writing-different-telemetry-types.md:157]: Let's remove the line breaks here

File: docs/features/writing-different-telemetry-types.md [hunk 4]
```diff
@@ -131,6 +131,48 @@ using (var measurement = DependencyMeasurement.Start("get-products"))
...
+
+```csharp
+var telemetryContext = new Dictionary<string, object>
+{
+    { "Local", "true" }
```
Comment (tomkerkhove) [inline docs/features/writing-different-telemetry-types.md:162]: Let's use a better example here for this demo:
```suggestion
    { "Subject", "Your order is being processed!" },
    { "OrderId", "ABC" }
```

File: src/Arcus.Observability.Tests.Integration/Serilog/ApplicationInsightsSinkTests.cs [hunk 5]
```diff
@@ -121,6 +121,30 @@ public void LogRequest_SinksToApplicationInsights_ResultsInRequestTelemetry()
...
+
+                string dependencyType = _bogusGenerator.Name.FullName();
+                object dependencyData = _bogusGenerator.Finance.Account();
+                bool isSuccessful = _bogusGenerator.PickRandom(true, false);
+                DateTimeOffset startTime = _bogusGenerator.Date.PastOffset();
+                TimeSpan duration = _bogusGenerator.Date.Timespan();
+                Dictionary<string, object> telemetryContext = CreateTestTelemetryContext();
+
+                // Act
+                logger.LogDependency(dependencyType, dependencyData, isSuccessful, startTime, duration, telemetryContext);
```
Comment (tomkerkhove) [inline src/Arcus.Observability.Tests.Integration/Serilog/ApplicationInsightsSinkTests.cs:141]: Did you verify it was tracked in Application Insights?

Comment (stijnmoreels) [reply src/Arcus.Observability.Tests.Integration/Serilog/ApplicationInsightsSinkTests.cs:141]: Yes, it was not tracked.

Comment (stijnmoreels) [reply src/Arcus.Observability.Tests.Integration/Serilog/ApplicationInsightsSinkTests.cs:141]: Just like the requests are not tracked.

Comment (tomkerkhove) [reply src/Arcus.Observability.Tests.Integration/Serilog/ApplicationInsightsSinkTests.cs:141]: Then we have to fix the tests before we merge it 🤷‍♂️

**Pull request merged**